### PR TITLE
Add Redpanda improvements

### PR DESCRIPTION
**Pull request by eddumelendez**
* Configure Redpanda via `.bootstrap.yml` and `redpanda.yml`
* Allow to enable authorization and authentication methods
* Allow to add additional listeners. E.g. using it with toxiproxy or redpanda console
* Enable rest proxy
* Configure rpk

See https://github.com/testcontainers/testcontainers-java/issues/6395

### CODE REVIEW

File: modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java [hunk 1]
```diff
@@ -96,6 +107,171 @@ public void testSchemaRegistry() {
...
+                    assertThatThrownBy(() -> adminClient.createTopics(topics).all().get(30, TimeUnit.SECONDS))
+                        .hasCauseInstanceOf(TopicAuthorizationException.class);
+                });
+        }
+    }
+
+    @Test
+    public void enableSaslAndWithAuthenticationError() {
+        try (
+            RedpandaContainer redpanda = new RedpandaContainer("docker.redpanda.com/redpandadata/redpanda:v23.1.7")
```
Comment (bochenekmartin) [inline modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java:184]: Redpanda `v23.2.2` has been recently released. Consider upgrading Redpanda version to the latest release

File: docs/modules/redpanda.md [hunk 2]
```diff
@@ -25,6 +25,43 @@ Redpanda also provides a schema registry implementation. Like the Redpanda broke
...
+<!--/codeinclude-->
+
+Below is an example of how to create the `AdminClient`:
+
+<!--codeinclude-->
+[Create Admin Client](../../modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java) inside_block:createAdminClient
+<!--/codeinclude-->
+
+There are scenarios where connecting to the broker is needed. The consumer/producer can be another container
+in the same network or a different process where port to connect to is different from the exposed port `9092`. E.g [Toxiproxy](../../docs/modules/toxiproxy.md).
```
Comment (kiview) [inline docs/modules/redpanda.md:47]:
```suggestion
in the same network or a different process where port to connect to differs from the default exposed port `9092`. E.g [Toxiproxy](../../docs/modules/toxiproxy.md).
```

File: docs/modules/redpanda.md [hunk 3]
```diff
@@ -25,6 +25,43 @@ Redpanda also provides a schema registry implementation. Like the Redpanda broke
...
+
+<!--codeinclude-->
+[Register listener](../../modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java) inside_block:registerListener
```
Comment (kiview) [inline docs/modules/redpanda.md:50]:
```suggestion
[Register additional listener](../../modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java) inside_block:registerListener
```

File: docs/modules/redpanda.md [hunk 4]
```diff
@@ -25,6 +25,43 @@ Redpanda also provides a schema registry implementation. Like the Redpanda broke
...
+<!--/codeinclude-->
+
+Container defined in the same network:
+
+<!--codeinclude-->
+[Create kcat container](../../modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java) inside_block:createKCatContainer
+<!--/codeinclude-->
+
+Client using the new listener registered:
```
Comment (kiview) [inline docs/modules/redpanda.md:59]:
```suggestion
Client using the new registered listener:
```

File: modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java [hunk 5]
```diff
@@ -47,33 +80,198 @@ public RedpandaContainer(DockerImageName imageName) {
...
+    public RedpandaContainer withSuperuser(String username) {
+        this.superusers.add(username);
+        return this;
+    }
+
+    /**
+     * Add a {@link Supplier} that will provide a listener with format {@code host:port}.
+     * Host will be added as a network alias.
+     * <p>
+     * The listener will be added to the existing listeners.
```
Comment (kiview) [inline modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java:190]:
```suggestion
     * The listener will be added to the default listeners.
```

File: modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java [hunk 6]
```diff
@@ -47,33 +80,198 @@ public RedpandaContainer(DockerImageName imageName) {
...
+     * <p>
+     * Existing listeners:
```
Comment (kiview) [inline modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java:192]:
```suggestion
     * Default listeners:
```

File: modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java [hunk 7]
```diff
@@ -47,33 +80,198 @@ public RedpandaContainer(DockerImageName imageName) {
...
+     * <ul>
+     *     <li>0.0.0.0:9092</li>
+     *     <li>0.0.0.0:9093</li>
+     * </ul>
+     * <p>
+     * Existing advertised listeners:
```
Comment (kiview) [inline modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java:198]:
```suggestion
     * Default advertised listeners:
```

**Review by kiview — APPROVED** [review]

**Review by bochenekmartin — APPROVED** [review]

**Pull request merged**